### PR TITLE
Add user creation and editing for admins

### PR DESCRIPTION
**Pull request by levinuss**
This commit enables admins to create new users and edit existing ones through the user management interface.

### CODE REVIEW

**Review by schu — CHANGES_REQUESTED** [review]
Looks good to me UI-wise. A couple of comments inline. Unless there's a good reason for why admins should be able to initiate a password reset, I would prefer if only the user personally can inititiate a reset.

File: resources/views/user.blade.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                    <div class="col-md-4 mb-3">
+                        <label for="created_at" class="form-label">Created</label>
+                        <input id="created_at" class="form-control" type="text" placeholder="{{ $user->created_at }}" disabled>
+                    </div>
+                    <div class="col-md-4 mb-3">
+                        <label for="updated_at" class="form-label">Updated</label>
+                        <input id="updated_at" class="form-control" type="text" placeholder="{{ $user->updated_at }}" disabled>
+                    </div>
+                    <div class="col-md-4 mb-3">
+                        <label for="email_verified_at" class="form-label">E-Mail verified</label>
```
Comment (schu) [inline resources/views/user.blade.php:91]: `E-Mail verified` -> `Email verified`

File: resources/views/user.blade.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+@section('title', 'Edit/New User')
+
+@section('content')
+<form id="submit-user" action="/admin/users/{{ isset($user) ? $user->id : '' }}" method="post" class="mt-3 mb-3">
+    @isset($user)
+        @method('PUT')
+    @endisset
+    @csrf
+</form>
```
[the remark this answers is not in the document]
Comment (levinuss) [reply resources/views/user.blade.php:11]: Form nesting is not allowed, so we would not be able to have the password reset form in this place. Or am I missing something?

**Review by schu — CHANGES_REQUESTED** [review]
Late second thought: let's not include the delete part for now, I think that requires more thought (what can and should be deleted, what not – for example, what should happen with _public-rw-listed_ decks) and work (for example, it doesn't work as-is if the user has created sessions due to a foreign key violation).

Also, two style nits inline.

File: resources/views/user.blade.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (schu) [reply resources/views/user.blade.php:11]: OK, I didn't remember that.

**Comment by schu** [comment]
Thanks!

Two thoughts for a later update:

- I would prefer if we restrict edits for users loaded from Keycloak (SSO), i.e. don't allow name and email edits (Keycloak is source of truth) and disable setting a password (users should not be able to bypass the SSO system)
- Creating a new user with an invalid email address results in the user being created and a `UnexpectedResponseException` being returned (`sendResetLink` fails)

**Pull request merged**